src/importer.ts

```diff
@@ -22,65 +24,75 @@ export interface ImporterConfig {
 
 export class Importer {
 
+    private batchNumber: number = 0;
+
     constructor(
         private papiClient: PapiClient,
         private config: ImporterConfig,
         private amqpConnectionProvider
     ) {}
 
-    public importAllEvents(): void {
-        let runCount = 0;
-        let amqpChannel;
+    public importAll(): void {
         this.amqpConnectionProvider
             .flatMap(connection => connection.createChannel())
-            .flatMap(channel => channel.assertQueue(this.config.amqp.queues.event, { durable: true }))
+            .flatMap(channel => Observable.merge(
```

Comment:
I guess you're trying to sync on observables completion. I would have used a combination operator like:
`Observable.zip(assertQ1, assertQ2)`
Or better (but never used!) `forkJoin` operator.
Not sure though :)